// @jahia/data-helper/dist/js/hooks/useTreeEntries/useTreeEntries.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTreeEntries = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _client = require("@apollo/client");
var _fragments = require("../../fragments/fragments.utils");
var _useTreeEntries = require("./useTreeEntries.gql-queries");
var _excluded = ["data"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var useTreeEntries = exports.useTreeEntries = function useTreeEntries(_ref, queryOptions) {
  var fragments = _ref.fragments,
    rootPaths = _ref.rootPaths,
    openPaths = _ref.openPaths,
    selectedPaths = _ref.selectedPaths,
    openableTypes = _ref.openableTypes,
    selectableTypes = _ref.selectableTypes,
    recursionTypesFilter = _ref.recursionTypesFilter,
    queryVariables = _ref.queryVariables,
    hideRoot = _ref.hideRoot,
    sortBy = _ref.sortBy;
  var query = (0, _fragments.replaceFragmentsInDocument)(_useTreeEntries.TREE_QUERY, fragments);
  var getTreeEntries = function getTreeEntries(data) {
    var treeEntries = [];
    var nodesById = {};
    var jcr = data ? data.jcr : {};
    var addNode = function addNode(node, depth, index) {
      var selected = false;
      if (node.selectable) {
        selected = selectedPaths.indexOf(node.path) !== -1;
      }
      var treeEntry = {
        name: node.name,
        path: node.path,
        open: node.openable && openPaths.indexOf(node.path) !== -1,
        selected: selected,
        openable: node.openable,
        selectable: node.selectable,
        depth: depth,
        prefix: '&nbsp;'.repeat(depth * 3),
        node: node,
        hidden: false,
        hasChildren: node.children.pageInfo.nodesCount > 0
      };
      treeEntries.splice(index, 0, treeEntry);
      nodesById[node.uuid] = treeEntry;
      return treeEntry;
    };
    if (jcr) {
      if (jcr.rootNodes) {
        jcr.rootNodes.forEach(function (rootNode) {
          var root = addNode(rootNode, 0, 0);
          root.hidden = hideRoot;
        });
      }
      if (jcr.openNodes) {
        (0, _toConsumableArray2["default"])(jcr.openNodes).sort(function (a, b) {
          return a.path.localeCompare(b.path);
        }).forEach(function (node) {
          var parent = nodesById[node.uuid];
          if (parent) {
            var parentIndex = treeEntries.indexOf(parent);
            (0, _toConsumableArray2["default"])(node.children.nodes).reverse().forEach(function (child) {
              addNode(child, parent.depth + 1, parentIndex + 1);
            });
          }
        });
      }
    }
    return treeEntries.filter(function (treeNode) {
      return !treeNode.hidden;
    });
  };
  var vars = _objectSpread({
    rootPaths: rootPaths,
    types: Array.from(new Set([].concat((0, _toConsumableArray2["default"])(openableTypes || []), (0, _toConsumableArray2["default"])(selectableTypes || [])))),
    recursionTypesFilter: recursionTypesFilter || {
      types: 'nt:base',
      multi: 'NONE'
    },
    selectable: selectableTypes,
    openable: openableTypes,
    openPaths: openPaths,
    sortBy: sortBy
  }, queryVariables);
  var _useQuery = (0, _client.useQuery)(query, _objectSpread(_objectSpread({}, queryOptions), {}, {
      variables: vars
    })),
    data = _useQuery.data,
    others = (0, _objectWithoutProperties2["default"])(_useQuery, _excluded);
  return _objectSpread({
    treeEntries: getTreeEntries(data)
  }, others);
};
