// @jahia/data-helper/dist/esm/hooks/useSiteInfo/useSiteInfo.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import { useQuery } from '@apollo/client';
import { SITE_INFO_QUERY } from "./useSiteInfo.gql-queries";
var adaptSiteInfo = function adaptSiteInfo(data) {
  if (data && (data.jcr || data.wsDefault)) {
    var res = data.jcr ? data.jcr.result : data.wsDefault.result;
    if (res) {
      return _objectSpread({}, res.site);
    }
  }
  return {
    languages: []
  };
};
export var useSiteInfo = function useSiteInfo(_ref) {
  var siteKey = _ref.siteKey,
    displayLanguage = _ref.displayLanguage,
    uiLanguage = _ref.uiLanguage;
  var variables = {
    path: '/sites/' + siteKey,
    displayLanguage: displayLanguage,
    skipUILanguage: true,
    uiLanguage: ''
  };
  if (uiLanguage !== undefined) {
    variables.skipUILanguage = false;
    variables.uiLanguage = uiLanguage;
  }
  var res = useQuery(SITE_INFO_QUERY, {
    variables: variables,
    errorPolicy: 'ignore'
  });
  return _objectSpread(_objectSpread({}, res), {}, {
    siteInfo: adaptSiteInfo(res.data)
  });
};
