// @jahia/data-helper/dist/js/fragments/PredefinedFragments.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.subNodesCount = exports.siteLanguages = exports.siteHomePage = exports.primaryNodeType = exports.parentNode = exports.operationSupport = exports.nodeTypeSubTypes = exports.nodeTypeDisplayName = exports.nodeCacheRequiredFields = exports.mimeTypes = exports.lockInfo = exports.isExternal = exports.installedModules = exports.getProperties = exports.displayableNode = exports.displayName = exports.contentRestrictions = exports.childNodeTypes = exports.canLockUnlock = exports.allProperties = exports.aggregatedPublicationInfo = exports.PredefinedFragments = void 0;
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _graphqlTag = _interopRequireDefault(require("graphql-tag"));
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7, _templateObject8, _templateObject9, _templateObject10, _templateObject11, _templateObject12, _templateObject13, _templateObject14, _templateObject15, _templateObject16, _templateObject17, _templateObject18, _templateObject19, _templateObject20, _templateObject21;
var displayName = exports.displayName = {
  variables: {
    language: 'String!'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["fragment DisplayName on JCRNode {\n        displayName(language:$language)\n    }"])))
};
var primaryNodeType = exports.primaryNodeType = {
  variables: {
    displayLanguage: 'String!',
    displayLanguageSet: 'Boolean!'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeInfoPrimaryNodeType on JCRNode {\n        primaryNodeType {\n            name\n            displayName(language: $displayLanguage) @include(if: $displayLanguageSet)\n            icon\n        }\n    }"])))
};
var parentNode = exports.parentNode = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["fragment ParentNodeInfo on JCRNode {\n        parent {\n            path\n            name\n            ...NodeCacheRequiredFields\n        }\n    }"])))
};
var aggregatedPublicationInfo = exports.aggregatedPublicationInfo = {
  variables: {
    language: 'String!',
    aggregatedPublicationInfoSubNodes: 'Boolean',
    aggregatedPublicationInfoIncludeReference: 'Boolean'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2["default"])(["fragment AggregatedPublicationInfo on JCRNode {\n        aggregatedPublicationInfo(language: $language, subNodes: $aggregatedPublicationInfoSubNodes, references:$aggregatedPublicationInfoIncludeReference) {\n            publicationStatus\n            existsInLive\n        }\n    }"])))
};
var operationSupport = exports.operationSupport = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2["default"])(["fragment OperationSupport on JCRNode {\n        operationsSupport {\n            lock\n            markForDeletion\n            publication\n        }\n    }"])))
};
var allProperties = exports.allProperties = {
  variables: {
    language: 'String!'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeAllProperties on JCRNode {\n        properties(language:$language) {\n            name\n            value\n            values\n        }\n    }"])))
};
var getProperties = exports.getProperties = {
  variables: {
    language: 'String!',
    getPropertiesNames: '[String!]!'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeProperties on JCRNode {\n        properties(names: $getPropertiesNames, language: $language) {\n            name\n            value\n            values\n        }\n    }"])))
};
var installedModules = exports.installedModules = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject8 || (_templateObject8 = (0, _taggedTemplateLiteral2["default"])(["fragment SiteInstalledModules on JCRNode {\n        site {\n            installedModulesWithAllDependencies\n            ...NodeCacheRequiredFields\n        }\n    }"])))
};
var siteLanguages = exports.siteLanguages = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject9 || (_templateObject9 = (0, _taggedTemplateLiteral2["default"])(["fragment SiteLanguages on JCRNode {\n        site {\n            defaultLanguage\n            ...NodeCacheRequiredFields\n            languages {\n                displayName\n                language\n                activeInEdit\n            }\n        }\n    }"])))
};
var displayableNode = exports.displayableNode = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject10 || (_templateObject10 = (0, _taggedTemplateLiteral2["default"])(["fragment DisplayableNodePath on JCRNode {\n        displayableNode {\n            path\n            ...NodeCacheRequiredFields\n        }\n    }"])))
};
var lockInfo = exports.lockInfo = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject11 || (_templateObject11 = (0, _taggedTemplateLiteral2["default"])(["fragment LockInfo on JCRNode {\n        lockOwner: property(name: \"jcr:lockOwner\") {\n            value\n        }\n        lockTypes: property(name: \"j:lockTypes\") {\n            values\n        }\n    }"])))
};
var canLockUnlock = exports.canLockUnlock = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject12 || (_templateObject12 = (0, _taggedTemplateLiteral2["default"])(["fragment CanLockUnlockInfo on JCRNode {\n        lockInfo {\n            canLock,\n            canUnlock,\n        }\n    }"])))
};
var subNodesCount = exports.subNodesCount = {
  variables: {
    subNodesCountTypes: '[String!]!'
  },
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject13 || (_templateObject13 = (0, _taggedTemplateLiteral2["default"])(["fragment SubNodesCount on JCRNode {\n        subNodes: children(typesFilter: {types: $subNodesCountTypes, multi: ANY}) {\n            pageInfo {\n                totalCount\n            }\n        }\n    }"])))
};
var childNodeTypes = exports.childNodeTypes = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject14 || (_templateObject14 = (0, _taggedTemplateLiteral2["default"])(["fragment AllowedChildNodeType on JCRNode {\n        allowedChildNodeTypes(includeSubTypes: false) {\n            name\n        }\n    }"])))
};
var contentRestrictions = exports.contentRestrictions = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject15 || (_templateObject15 = (0, _taggedTemplateLiteral2["default"])(["fragment ContentRestriction on JCRNode {\n        contributeTypes: property(name: \"j:contributeTypes\") {\n            values\n        }\n        ancestors(fieldFilter: {filters: {evaluation: NOT_EMPTY, fieldName: \"contributeTypes\"}}) {\n            ...NodeCacheRequiredFields\n            contributeTypes : property(name: \"j:contributeTypes\") {\n                values\n            }\n        }\n    }"])))
};
var siteHomePage = exports.siteHomePage = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject16 || (_templateObject16 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeSiteHomePage on JCRNode {\n        children(typesFilter:{types:[\"jnt:page\"]}, propertiesFilter:{filters:[{property:\"j:isHomePage\", value:\"true\" }]}) {\n            nodes {\n                path\n                ...NodeCacheRequiredFields\n            }\n        }\n    }"])))
};
var nodeCacheRequiredFields = exports.nodeCacheRequiredFields = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject17 || (_templateObject17 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeCacheRequiredFields on JCRNode {\n        uuid\n        workspace\n        path\n    }"])))
};
var nodeTypeDisplayName = exports.nodeTypeDisplayName = {
  variables: {
    language: 'String!'
  },
  applyFor: 'nodeType',
  gql: (0, _graphqlTag["default"])(_templateObject18 || (_templateObject18 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeTypeDisplayName on JCRNodeType {\n        displayName(language:$language)\n    }"])))
};
var nodeTypeSubTypes = exports.nodeTypeSubTypes = {
  variables: {
    language: 'String!'
  },
  applyFor: 'nodeType',
  gql: (0, _graphqlTag["default"])(_templateObject19 || (_templateObject19 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeTypeSubTypes on JCRNodeType {\n        subTypes {\n            nodes {\n                name\n                displayName(language:$language)\n                abstract\n                mixin\n            }\n        }\n    }"])))
};
var mimeTypes = exports.mimeTypes = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject20 || (_templateObject20 = (0, _taggedTemplateLiteral2["default"])(["fragment NodeInfoResourceNode on JCRNode {\n        resourceChildren: children(names: [\"jcr:content\"]) {\n            nodes {\n                ...NodeCacheRequiredFields\n                mimeType: property(name: \"jcr:mimeType\") {\n                    value\n                }\n            }\n        }\n    }"])))
};
var isExternal = exports.isExternal = {
  applyFor: 'node',
  gql: (0, _graphqlTag["default"])(_templateObject21 || (_templateObject21 = (0, _taggedTemplateLiteral2["default"])(["fragment External on JCRNode {\n        isExternal\n    }"])))
};
var PredefinedFragments = exports.PredefinedFragments = {
  displayName: displayName,
  primaryNodeType: primaryNodeType,
  parentNode: parentNode,
  aggregatedPublicationInfo: aggregatedPublicationInfo,
  operationSupport: operationSupport,
  allProperties: allProperties,
  getProperties: getProperties,
  installedModules: installedModules,
  siteLanguages: siteLanguages,
  displayableNode: displayableNode,
  lockInfo: lockInfo,
  subNodesCount: subNodesCount,
  contentRestrictions: contentRestrictions,
  siteHomePage: siteHomePage,
  nodeCacheRequiredFields: nodeCacheRequiredFields,
  nodeTypeDisplayName: nodeTypeDisplayName,
  nodeTypeSubTypes: nodeTypeSubTypes,
  mimeTypes: mimeTypes,
  isExternal: isExternal
};
