// @jahia/data-helper/dist/js/hooks/useNodeChecks/useNodeChecks.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useNodeChecks = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _ = require("./..");
var _excluded = ["requiredPermission", "requiredSitePermission", "showOnNodeTypes", "hideOnNodeTypes", "requireModuleInstalledOnSite", "showForPaths", "hideForPaths", "hideOnExternal"],
  _excluded2 = ["node", "nodes", "loading"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var evaluateVisibilityPaths = function evaluateVisibilityPaths(visible, visibilityPaths, nodePath) {
  for (var i = 0; i < visibilityPaths.length; i++) {
    if (new RegExp(visibilityPaths[i]).test(nodePath)) {
      return visible;
    }
  }
  return !visible;
};
function addArrayOptionValues(newValue, useNodeInfoOptions, useNodeInfoKey) {
  if (newValue) {
    useNodeInfoOptions[useNodeInfoKey] = useNodeInfoOptions[useNodeInfoKey] || [];
    useNodeInfoOptions[useNodeInfoKey] = useNodeInfoOptions[useNodeInfoKey].concat(newValue.filter(function (item) {
      return useNodeInfoOptions[useNodeInfoKey].indexOf(item) < 0;
    }));
  }
}
var useNodeChecks = exports.useNodeChecks = function useNodeChecks(variables, options, queryOptions) {
  var requiredPermission = options.requiredPermission,
    requiredSitePermission = options.requiredSitePermission,
    showOnNodeTypes = options.showOnNodeTypes,
    hideOnNodeTypes = options.hideOnNodeTypes,
    requireModuleInstalledOnSite = options.requireModuleInstalledOnSite,
    showForPaths = options.showForPaths,
    hideForPaths = options.hideForPaths,
    hideOnExternal = options.hideOnExternal,
    othersOptions = (0, _objectWithoutProperties2["default"])(options, _excluded);
  var useNodeInfoOptions = _objectSpread({}, othersOptions);
  var requiredPermissions = typeof requiredPermission === 'string' ? [requiredPermission] : requiredPermission;
  var requiredSitePermissions = typeof requiredSitePermission === 'string' ? [requiredSitePermission] : requiredSitePermission;
  addArrayOptionValues(requiredPermissions, useNodeInfoOptions, 'getPermissions');
  addArrayOptionValues(requiredSitePermissions, useNodeInfoOptions, 'getSitePermissions');
  addArrayOptionValues(showOnNodeTypes, useNodeInfoOptions, 'getIsNodeTypes');
  addArrayOptionValues(hideOnNodeTypes, useNodeInfoOptions, 'getIsNodeTypes');
  if (requireModuleInstalledOnSite) {
    useNodeInfoOptions.getSiteInstalledModules = true;
  }
  var _useNodeInfo = (0, _.useNodeInfo)(variables, useNodeInfoOptions, queryOptions),
    node = _useNodeInfo.node,
    nodes = _useNodeInfo.nodes,
    loading = _useNodeInfo.loading,
    othersResults = (0, _objectWithoutProperties2["default"])(_useNodeInfo, _excluded2);
  if (loading) {
    return _objectSpread({
      loading: loading
    }, othersResults);
  }
  if (!node && !nodes) {
    return _objectSpread({
      checksResult: false,
      loading: loading
    }, othersResults);
  }
  var doNodeCheck = function doNodeCheck(currentNode) {
    return (!requiredPermissions || requiredPermissions.reduce(function (acc, val) {
      return acc || currentNode[val];
    }, false)) && (!requiredSitePermissions || requiredSitePermissions.reduce(function (acc, val) {
      return acc || currentNode.site[val];
    }, false)) && (!showOnNodeTypes || showOnNodeTypes.reduce(function (acc, val) {
      return acc || currentNode[val];
    }, false)) && (!hideOnNodeTypes || !hideOnNodeTypes.reduce(function (acc, val) {
      return acc || currentNode[val];
    }, false)) && (!requireModuleInstalledOnSite || requireModuleInstalledOnSite.reduce(function (acc, val) {
      return acc && currentNode.site.installedModulesWithAllDependencies.includes(val);
    }, true)) && (!hideForPaths || evaluateVisibilityPaths(false, hideForPaths, currentNode.path || variables.path)) && (!showForPaths || evaluateVisibilityPaths(true, showForPaths, currentNode.path || variables.path)) && (!hideOnExternal || !currentNode.isExternal);
  };
  var result = node ? doNodeCheck(node) : nodes.reduce(function (acc, val) {
    return acc && doNodeCheck(val);
  }, true);
  return _objectSpread({
    node: node,
    nodes: nodes,
    checksResult: result,
    loading: loading
  }, othersResults);
};
